### PR TITLE
CI Improvements

### PR DESCRIPTION
**Pull request by s0**
Further to the CI integrations mentioned in #13 (prettier & eslint), this PR makes a few more improvements:

* Also check that package-lock.json is unchanged when running NPM install in Node.js v16 (this will fail when packages are installed and this file isn't updated, or when the version in `package.json` is updated but `package-lock.json` is not updated to match.
* Run TypeScript compilation checks to make sure that everything type-checks.
* Run Push & PR checks regardless of branch (I was finding it frustrating trying to determine whether what I'd written was working correctly on my fork until I had done this, and I imagine many others will find the same).

Open to discussing any of the above of course :)

Results when invalid TS or `package-lock.json` are pushed can be found here: https://github.com/s0/nody-greeter/actions/runs/1830680405

### CODE REVIEW

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -1,16 +1,45 @@
...
+        # rerun when typescript files change
       - "ts/**"
-  pull_request:
-    branches: [ master ]
+        # rerun when workflow is changed
+      - ".github/workflows/check.yml"
+        # rerun when packages change
+      - "package.json"
+      - "package-lock.json"
+  pull_request: {}
```
Comment (JezerM) [inline .github/workflows/check.yml:13]:
```suggestion
  pull_request:
```

Is the empty dict needed?

**Review by JezerM — COMMENTED** [review]
Overall, looks nice~

Thanks for this PR!

File: .github/workflows/check.yml (hunk 1 above)
Comment (s0) [reply .github/workflows/check.yml:13]: Probably not actually! I'll try it without!

Comment (s0) [reply .github/workflows/check.yml]: Yep seems to work fine!

**Review by JezerM — APPROVED** [review]
Nice~

**Pull request merged**